vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java

```diff
@@ -132,6 +139,57 @@ public String toString()
         });
     }
 
+    /**
+     * Waits for an <b><i>element</i></b> with the specified <b>locator</b> to stop moving
+     * @param locator locator to locate element
+     */
+    @When("I wait until element located by `$locator` stops moving")
+    public void waitForElementStopsMoving(Locator locator)
+    {
+        uiContext.getOptionalSearchContext().ifPresent(context -> {
+            locator.getSearchParameters().setWaitForElement(false);
+            IExpectedSearchContextCondition<Boolean> condition = new IExpectedSearchContextCondition<>() {
+                private Point lastPosition;
+                private boolean isElementPresent = true;
+
+                @Override
+                public Boolean apply(SearchContext context)
+                {
+                    try
+                    {
+                        return searchActions.findElement(context, locator).map(WebElement::getLocation)
+                                .map(currentPosition ->
+                                {
+                                    boolean samePosition = currentPosition.equals(lastPosition);
+                                    LOGGER.info("Detected location of moving element: '{}'", currentPosition);
+                                    lastPosition = currentPosition;
+                                    isElementPresent = true;
+                                    return samePosition;
+                                }).orElse(false);
+                    }
+                    catch (StaleElementReferenceException e)
+                    {
+                        if (!isElementPresent)
+                        {
+                            throw new NoSuchElementException("No such element located by \"" + locator
+                                    .toHumanReadableString() + "\". ", e);
```

Comment:
actually this error is misleading, the element was present, but disappeared

Reply:
changed